### ISSUE TITLE
tab stops not working?

### ISSUE THREAD

**Issue by guidorice** [unicodeveloper/atom-aurelia-snippets#1]
First off thanks so much for creating this snippets package! It's helping me learn Aurelia.

One thing I have noticed, is that in my ES6 project, the tab stops do not seem work. I was investigating this and noticed the way you have annotated the tabs stops is slightly different than shown here:

http://flight-manual.atom.io/using-atom/sections/snippets/

Also I noticed some of the javascript snippets use Typescript syntax. Maybe you could provide an option in the package config to select ESNext or Typescript syntax?

using: Atom 1.9.9

Let me know if you have any suggestions- I would be happy to fork, if that would be best way to proceed.

Cheers

**Comment by unicodeveloper** [comment]
Hi @guidorice I'm sorry you are experiencing this, but none of the javascript snippets use Typescript.  I'll be happy if you can point me to anyone that looks like typescript

**Comment by guidorice** [comment]
Hi, are the tab stops working for you? Maybe it's just a problem with my setup- they have never worked at all, but just for these aurelia snippets.

re: the typescript, I thought that what I was seeing in some places was type definitions, for example:

```
"Aurelia: created(ov, mv) { ... }":
    prefix: "_cre"
    body: "created(owningView: View, myView: View) {\n\t${1}\n}${2}"
```

So `: View` is the type definition for TS, right?

**Comment by unicodeveloper** [comment]
Oh, that's just Aurelia way of typehinting the parameters. It perfectly works for me and others that have tried it. What version of atom are you using?

**Comment by guidorice** [comment]
Ah- weird! I am using 1.9.9 on OS X. For example if I type `_req [ENTER]` it fills in `<require from="${url}"></require>` and the cursor is after the closing tag. Hitting tab just inserts tabs to the right of there.

**Comment by unicodeveloper** [comment]
Enter or tab should work fine. So Enter works for you?

**Comment by guidorice** [comment]
No ENTER has the same result as if I hit TAB.

**Comment by unicodeveloper** [comment]
Can you try this on another machine?

**Comment by guidorice** [comment]
Sure, here are steps I took to repro, this time on another machine (Linux Ubuntu 16.04)
- Upgraded Atom to latest 1.9.9
- Disabled all other packages in Atom, except for aurelia-snippets, to make sure no conflicts come up.
- `npm install aurelia-cli -g`
- `au new test-prj`

```
Project Configuration
    Name: test-prj
    Platform: Web
    Transpiler: Babel
    CSS Processor: None
    Unit Test Runner: Karma
    Editor: Atom
```
- `cd test-prj/`
- `atom .`
- open src/app.html
- type _req[TAB]
- snippet fills in `<require from="${url}"></require>` with cursor at end of line. The tab stop does not occur.
- confirmed same for other snippets, like `_rpf` and also confirmed ENTER has same behavior as TAB.

**Comment by guidorice** [comment]
I also nuked my ~/.atom config, and started with vanilla settings, re-installed aurelia-snippets, and confirmed the same results again.

**Comment by unicodeveloper** [comment]
Is this still an issue?

**Comment by guidorice** [comment]
Yes, still does not work. I have not had time to debug it any more, beyond what I already posted.
